src/imagej/__init__.py

```diff
@@ -1124,6 +1125,7 @@ def init(
     ij_dir_or_version_or_endpoint=None,
     mode: Union[Mode, str] = Mode.HEADLESS,
     add_legacy=True,
+    override=False,
```

Comment:
Seeing this parameter right next to the deprecated `headless` parameter makes me wonder whether we are over-eager to add parameters to this method.  

Furthermore, the docstring is necessary to understand *what* is being overridden. I think we could do better.

What do you think about adding an environment variable check, which would obscure the override a bit more? We could set the environment variable [here](https://github.com/apposed/jaunch/blob/392c60e690fa19c4c12b9003696ad334fbad4dec/configs/fiji.py#L63) (maybe there's a better place, I don't know jaunch very well)

Reply:
I agree about being hesitant about adding a new parameter to the `init()` method. Honestly, I would much prefer to **not** add a parameter. I only did so after my attempts at trying to discover if the current Python instance was started with `pthread` (i.e. `jaunch` with python interactive mode enabled). Long story short, I don't think Python can determine _how_ it was launched. 

> What do you think about adding an environment variable check, which would obscure the override a bit more? We could set the environment variable [here](https://github.com/apposed/jaunch/blob/392c60e690fa19c4c12b9003696ad334fbad4dec/configs/fiji.py#L63) (maybe there's a better place, I don't know jaunch very well)

Well I don't think knowing that we are jaunching from macOS helps here. The problem is that we have [this check](https://github.com/imagej/pyimagej/blob/55dbfda7aae5469471bc1f52b7c6df87d67e98d3/src/imagej/__init__.py#L1205-L1208) for interactive mode on macOS that raises if `True`. What do you think about adding a new `mode` like "jaunch" instead of a new parameter?

Reply:
> I only did so after my attempts at trying to discover if the current Python instance was started with pthread (i.e. jaunch with python interactive mode enabled) on macOS. 

[This SO post](https://stackoverflow.com/q/4867839/1207769) goes into some platform-specific ways of detecting whether the current pthread is the main thread. But one idea written there gave me an idea for an easier solution: Have Jaunch set an environment variable or other telltale somewhere that our Python code can then check. Since we control both our Python code and the Jaunch code, we should be able to "communicate with ourselves" when Jaunch launches Python on a non-main pthread. It's not a perfectly general solution, but better than nothing?

Reply:
> The problem is that we have [this check](https://github.com/imagej/pyimagej/blob/55dbfda7aae5469471bc1f52b7c6df87d67e98d3/src/imagej/__init__.py#L1205-L1208) for interactive mode on macOS that raises if True.

I thought the plan was to remove that check now, in favor of a smarter check. We shouldn't need a new mode nor any new boolean `override` flag.

Reply:
This is the plan. I fell back to adding an `override` flag because I couldn't find a way to determine how a Python instanced was launched. And I agree its an unfavorable solution. I haven't seen the SO post you linked so I think trying what they suggest there or perhaps as you suggest, setting an environment variable, is the way to go :+1: .

Reply:
Good news, importing the pthread C library as suggested by the SO post works! I will remove the override flag and implement this check in the next wave of commits. Here's how you check if the current thread is the main thread or non-main thread (_i.e._ jaunched).

```python
from ctypes import cdll

# macOS uses linked cache system for dylib files -- you just need the correct name
pthread = cdll.LoadLibrary("libpthread.dylib")

# check if main thread
print(pthread.pthread_main_np())
```

This will return `1` if the current thread is the main thread and `0` if its not. You can also use `libSystem.dylib` instead of `libpthread.dylib`. Because of the linked cache system uses you wont find these files actually on disk (a source of my confusion for a while...UGH).

Reply:
The `override` flag has been removed now.